### PR TITLE
feat(argo-cd): Allow setting the private key algorithm in the Certificate spec

### PR DESCRIPTION
**Pull request by xaque208**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -1007,6 +1007,9 @@ server:
     duration: ""
     # -- How long before the currently issued certificate's expiry cert-manager should renew the certificate. Value must be in units accepted by Go time.ParseDuration
     renewBefore: ""
+    # -- The private key algorithm to use in the Certificate spec.
+    privateKey:
+      algorithm: ""
```
Comment (pdrastil) [inline charts/argo-cd/values.yaml:1012]: How about exposing whole [CertificatePrivateKey](https://cert-manager.io/docs/reference/api-docs/#cert-manager.io/v1.CertificatePrivateKey) spec like below with sensible defaults?

```yaml
# Private key of the certificate
privateKey:
  # -- Rotation policy of private key when certificate is re-issued. Either: `Never` or `Always`
  rotationPolicy: Never
  # -- The private key cryptography standards (PKCS) encoding for private key. Either: `PCKS1` or `PKCS8`
  encoding: PKCS1
  # -- Algorithm used to generate certificate private key. One of: `RSA`, `Ed25519` or `ECDSA`
  algorithm: RSA
  # -- Key bit size of the private key. If algorithm is set to `Ed25519`, size is ignored.
  size: 2048
```

File: charts/argo-cd/templates/argocd-server/certificate.yaml [hunk 2]
```diff
@@ -31,6 +31,10 @@ spec:
     {{- end }}
     kind: {{ .Values.server.certificate.issuer.kind | quote }}
     name: {{ .Values.server.certificate.issuer.name | quote }}
+  privateKey:
```
Comment (pdrastil) [inline charts/argo-cd/templates/argocd-server/certificate.yaml:34]: Please check for the whole specification of the privateKey below and use following construct:

```yaml
{{- with .Values.server.certificate.privateKey }}
privateKey:
  {{- toYaml . | nindent 4 }}
{{- end }}
```

**Review by pdrastil — CHANGES_REQUESTED** [review]
Hi,
thanks for contribution. Please take a look on suggested changes.

File: charts/argo-cd/Chart.yaml [hunk 3]
```diff
@@ -20,3 +20,4 @@ dependencies:
 annotations:
   artifacthub.io/changes: |
     - "[Fixed]: Additional initContainers are handled correctly in the dex deployment"
```
[the remark this answers is not in the document]
Comment (xaque208) [reply charts/argo-cd/Chart.yaml:22]: Correct.  It was there before I made the change.  Is there another way I should note the changes?

[the remark this answers is not in the document]
Comment (xaque208) [reply charts/argo-cd/Chart.yaml]: I've scrubbed the old entries during the rebase.

File: charts/argo-cd/values.yaml (hunk 1 above)
Comment (xaque208) [reply charts/argo-cd/values.yaml]: This has been updated with your recommendations.  Good idea.

File: charts/argo-cd/templates/argocd-server/certificate.yaml (hunk 2 above)
Comment (xaque208) [reply charts/argo-cd/templates/argocd-server/certificate.yaml:35]: I've updated to wrap all of the `privateKey` as suggested.

File: charts/argo-cd/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (pdrastil) [reply charts/argo-cd/Chart.yaml]: Thanks - the changes is the release log for the given version.

**Review by pdrastil — APPROVED** [review]
LGTM

**Comment by pdrastil** [comment]
@xaque208 Thanks for the changes. Please update the docs via `scripts/helm-docs.sh` so the build will pass and this can be merged.

**Pull request merged**